datafusion/sql/src/utils.rs

```diff
@@ -542,3 +542,11 @@ pub(crate) fn normalize_ident(id: &Ident) -> String {
         None => id.value.to_ascii_lowercase(),
     }
 }
+
+// Normalize an owned identifier to a lowercase string unless the identifier is quoted.
```

Comment:
I will remove the non-owned variant in a follow on PR -- but I was trying to minimize the size of this one

Reply:
https://github.com/apache/arrow-datafusion/pull/4534#